data_processing/meteo/pg_processing/task_functions.py

```diff
@@ -0,0 +1,286 @@
+                            table_name = config[dataset]["table_name"]
+                        mydict[str(DATADIR  + table_name + "/" + res["title"] + ".csv")] = {"name": file_path.name, "regex_infos": regex_infos}
+            else:
+                file_path = download_resource(res, dataset)
+                regex_infos = get_regex_infos(config[dataset]["source_pattern"], res["url"].split("/")[-1], config[dataset]["params"])
+                if dataset == "BASE/QUOT":
+                    if "_autres" in file_path.name:
+                        table_name = "base_quot_autres"
+                    else:
+                        table_name = "base_quot_vent"
+                else:
+                    table_name = config[dataset]["table_name"]
+                mydict[DATADIR  + table_name + "/" + res["title"] + ".csv"] = {"name": file_path.name, "regex_infos": regex_infos}
+    return mydict
+
+def download_data(ti):
+    latest_processed_date = ti.xcom_pull(key="latest_processed_date", task_ids="retrieve_latest_processed_date")
+    latest_ftp_processing = ti.xcom_pull(key="latest_ftp_processing", task_ids="get_latest_ftp_processing")
+    mydict = {}
+    if not latest_processed_date:
+        # Process everything
+        for dataset in DATASETS_TO_PROCESS:
+            print(f"getting all batch for {dataset}")
+            resources = fetch_resources(dataset)
+            mydict.update(process_resources(resources[:4], dataset, latest_ftp_processing))
+        yesterday = (datetime.today() - timedelta(days=1)).strftime('%Y-%m-%d')
+        ti.xcom_push(key="latest_processed_date", value=yesterday)
+    else:
+        dates = [item for item in latest_ftp_processing if item != 'latest_update']
+        dates = [item for item in dates if item > latest_processed_date]
+        for dataset in DATASETS_TO_PROCESS:
+            resources = fetch_resources(dataset)
+            mydict.update(process_resources(resources, dataset, latest_ftp_processing, dates))
+        ti.xcom_push(key="latest_processed_date", value=max(dates))
+
+    ti.xcom_push(key="regex_infos", value=mydict)
+
+
+
+def unzip_csv_gz():
+    for dataset in DATASETS_TO_PROCESS:
+        patterns = []
+        if dataset == "BASE/QUOT":
+            pattern_vent = os.path.join(DATADIR, config[dataset]["table_name"] + "_vent", '*.csv.gz')
+            pattern_autres = os.path.join(DATADIR, config[dataset]["table_name"] + "_autres", '*.csv.gz')
+            patterns.extend([pattern_vent, pattern_autres])
+        else:
+            pattern = os.path.join(DATADIR, config[dataset]["table_name"], '*.csv.gz')
+            patterns.append(pattern)
+
+        for pattern in patterns:
+            for file_path in glob.glob(pattern):
+                output_file_path = file_path[:-3]
+
+                with gzip.open(file_path, 'rb') as f_in:
+                    with open(output_file_path, 'wb') as f_out:
+                        shutil.copyfileobj(f_in, f_out)
```

Comment:
Can we delete the czv.gz after unzipping?
```suggestion
                        shutil.copyfileobj(f_in, f_out)
                os.remove(file_path)
```

Reply:
yes